src/main/java/jenkins/plugins/openstack/compute/SectionDisabled.java

```diff
@@ -0,0 +1,277 @@
+package jenkins.plugins.openstack.compute;
+
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import hudson.Extension;
+import hudson.Functions;
+import hudson.Util;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import hudson.util.FormValidation;
+
+/**
+ * Records that the user has disabled something "until further notice", or the
+ * system has disabled something for a period, or both.
+ */
+public class SectionDisabled extends AbstractDescribableImpl<SectionDisabled> implements Serializable {
```

Comment:
I am wondering if there is a way to implement this class immutable.

Reply:
I don't think this is a good candiate for being immutable.
When we have a plugin-imposed temporary disablement, that's a temporary state that'll time out; we don't really want to have the class be immutable as it'd mean we'd have to have a new instance.
Personally, I'd prefer a mutable instance to having to swap out different instances.